dvc/utils/__init__.py

```diff
@@ -152,9 +152,9 @@ def makedirs(path, exist_ok=False, mode=None):
         _makedirs(path, exist_ok=exist_ok)
         return
 
-    umask = os.umask(0)
+    umask = os.umask(0o777 - mode)
```

Comment:
Won't we get TypeError when mode is default?

Reply:
no, please take a look a few lines above - https://github.com/iterative/dvc/pull/2790/files#diff-150500aeed547602ede602be74a28df6R151-R153

Reply:
Ah sorry, didn't notice that.